src/Microsoft.Extensions.Configuration.AzureAppConfiguration/ConfigurationClients/FailOverSupportedConfigurationClient.cs

```diff
@@ -0,0 +1,256 @@
+                    tasks.Add(parallelAwait);
+
+                    var completedTask = await Task.WhenAny(tasks);
+                    if (!completedTask.Equals(parallelAwait))
+                    {
+                        if (completedTask.Status == TaskStatus.RanToCompletion)
+                        {
+                            client.BackoffEndTime = DateTimeOffset.UtcNow;
+                            client.FailedAttempts = 0;
+
+                            // Safe because the task is completed.
+                            return funcTask.Result;
+                        }
+                        else if (completedTask.Status == TaskStatus.Faulted)
+                        {
+                            tasks.Remove(completedTask);
+                            tasks.Remove(parallelAwait);
+
+                            if (IsRetryableException(completedTask.Exception))
+                            {
+                                lastException = completedTask.Exception;
+                                client.FailedAttempts++;
+                                TimeSpan backoffInterval = BackoffIntervalConstants.MinBackoffInterval.CalculateBackoffInterval(BackoffIntervalConstants.MaxBackoffInterval, client.FailedAttempts);
+                                client.BackoffEndTime = DateTimeOffset.UtcNow.Add(backoffInterval);
+                            }
+                            else
+                            {
+                                completedTask.Exception.Handle(e =>
+                                {
+                                    // propagate the original exception.
+                                    throw e;
+                                });
+                            }
+                        }
+                        else if (completedTask.Status == TaskStatus.Canceled)
+                        {
+                            // Throw expected OperationCanceledException
+                            completedTask.GetAwaiter().GetResult();
+                        }
+                    }
+                    else
+                    {
+                        tasks.Remove(parallelAwait);
+                    }
+                }
+            }
+
+            throw lastException;
+        }
+
+        private bool IsRetryableException(AggregateException ex)
+        {
+            int statusCode = 0;
+
+            IReadOnlyCollection<Exception> innerExceptions = ex.Flatten().InnerExceptions;
+
+            if (innerExceptions.Count > 0 && innerExceptions.All(ex => ex is RequestFailedException))
+            {
+                statusCode = (innerExceptions.Last() as RequestFailedException).Status;
+            }
+
+            switch (statusCode)
+            {
+                case (int)HttpStatusCode.RequestTimeout:       // 408
+                case HttpStatusCodeRequestThrottled:           // 429
+                case (int)HttpStatusCode.InternalServerError:  // 500
```

Comment:
As discussed we should fail over for all server errors, so why don't we just check the status code is equal to or greater than HttpStatusCode.InternalServerError?

Reply:
Like I mentioned in another comment, this is not about retry. This is failover. They are not exactly the same.

Reply:
I think it helps to make our failover strategy clearer.

(Let's close on 408 in the other discussion.)

Reply:
Discussed with Jimmy, we need to failover for HTTP 408.